jwt/api.py

```diff
@@ -110,14 +129,16 @@ def encode(self, payload, key, algorithm='HS256', headers=None, json_encoder=Non
 
         return b'.'.join(segments)
 
-    def decode(self, jwt, key='', verify=True, algorithms=None, **kwargs):
+    def decode(self, jwt, key='', verify=True, algorithms=None, options=None, **kwargs):
         payload, signing_input, header, signature = self._load(jwt)
 
         if verify:
-            self._verify_signature(payload, signing_input, header, signature,
-                                   key, algorithms)
+            merged_options = self._merge_options(override_options=options)
+            if merged_options.get('verify_signature'):
+                self._verify_signature(payload, signing_input, header, signature,
+                                       key, algorithms)
```

Comment:
Looking at the ruby library that the original issue was based on, it looks like they use the `verify` kwarg instead of `verify_signature` in the options dictionary at all.  

I can see taking it out of the dictionary completely to be the same as the ruby library or doing something like this:

```python
merged_options = self._merge_options(override_options=options)
if verify or merged_options.get('verify_signature'):
    self._verify_signature(payload, signing_input, header, signature,
                                       key, algorithms)

self._validate_claims(payload, options=merged_options, **kwargs)
```

This would avoid the `verify` kwarg from overriding the other options and require that the developer double opt-in to not verifying the signature.